common/static/common/js/spec/components/search_field_spec.js

```diff
@@ -0,0 +1,98 @@
+                    url: mockUrl,
+                    count: results.length,
+                    num_pages: results.length / pageSize,
+                    current_page: 1,
+                    start: 0,
+                    results: _.first(results, perPage)
+                },
+                {parse: true}
+            );
+            collection.start = 0;
+            collection.totalCount = results.length;
+            return collection;
+        };
+
+        var createSearchFieldView = function (options) {
+            options = _.extend(
+                {
+                    type: 'test',
+                    collection: newCollection(5, 4),
+                    el: $('.test-search')
+                },
+                options || {}
+            );
+            return new SearchFieldView(options || {});
+        };
+
+        beforeEach(function() {
+            setFixtures('<section class="test-search"></section>');
+        });
+
+        it('correctly displays itself', function () {
+            searchFieldView = createSearchFieldView({
+                collection: newCollection(20, 5)
+            }).render();
+            expect(searchFieldView.$('.search-field').val(), '');
+        });
+
+        it('can display with an initial search string', function () {
+            searchFieldView = createSearchFieldView({
+                searchString: 'foo'
+            }).render();
+            expect(searchFieldView.$('.search-field').val(), 'foo');
+        });
+
+        it('refreshes the collection when performing a search', function () {
+            var requests = AjaxHelpers.requests(this);
+            searchFieldView = createSearchFieldView().render();
+            searchFieldView.$('.search-field').val('foo');
+            searchFieldView.$('.action-search').click();
+            AjaxHelpers.expectRequestURL(requests, mockUrl, {
+                page: '1',
+                page_size: '10',
+                sort_order: '',
+                text_search: 'foo'
+            });
+            AjaxHelpers.respondWithJson(requests, {
+                count: 10,
+                current_page: 1,
+                num_pages: 1,
+                start: 0,
+                results: []
+            });
+            expect(searchFieldView.$('.search-field').val(), 'foo');
+        });
+
+        it('does not refresh when the server throws an error', function () {
+            var requests = AjaxHelpers.requests(this);
+            searchFieldView = createSearchFieldView().render();
+            searchFieldView.$('.search-field').val('foo');
+            searchFieldView.$('.action-search').click();
+            AjaxHelpers.expectRequestURL(requests, mockUrl, {
+                page: '1',
+                page_size: '10',
+                sort_order: '',
+                text_search: 'foo'
+            });
+            AjaxHelpers.respondWithError(requests);
+            expect(searchFieldView.$('.search-field').val(), 'foo');
```

Comment:
How can you tell that it didn't refresh? You are testing exactly the same things as on line 80.